angr/knowledge_plugins/labels.py

```diff
@@ -1,66 +1,237 @@
+            return namespace.setdefault(default, addr)
+
+        else:
+            return namespace.inv.get(addr)
+
+    def del_label(self, name, ns=_global_ns):
+        """Delete a label that is present within the given namespace.
+        
+        :param name:    The name of the label which is to be deleted.
+        :param ns:      The namespace in which the given label is present.
+        :return: 
+        """
+        namespace = self._namespaces[ns]  # a shorthand
+
+        if name not in namespace:
+            raise ValueError("Namespace '%s' doesn't have a label named '%s'" % (ns, name))
+
+        del namespace[name]
+
+    def iter_labels(self, addr, ns_set=None):
+        """Iterate over labels that are assigned to a given address in the given set of namespaces.
+        
+        :param addr:    An address for which to yield assigned labels.
+        :param ns_set:  A set of namespaces to work with.
+        :return:        Tuples of (namespace, label).
+        """
+        ns_set = self._normalize_ns_set(ns_set)
+
+        for ns in ns_set:
+            yield ns, self.get_label(addr, ns)
+
+    #
+    #   ...
+    #
+
+    def get_addr(self, name, ns=_global_ns, default=None):
+        """Get the address for the label in the given namespace.
+        
+        :param name:    The label name.
+        :param ns:      The namespace to look into.
+        :param default: Default value if the label is not present in the given namespace.
+        :return: 
+        """
+        namespace = self._namespaces[ns]  # a shorthand
+
+        return namespace.get(name, default)
+
+    def del_addr(self, addr, ns=_global_ns):
+        """Delete a label that is assigned to the given address within the given namespace.
+
+        :param addr:    The address from which to remove a label in the given namespace.
+        :param ns:      The namespace in which the given address is present.
+        :return: 
+        """
+        namespace = self._namespaces[ns]  # a shorthand
+
+        if addr not in namespace.inv:
+            raise ValueError("Namespace '%s' doesn't have a label on address %#x" % (ns, addr))
+
+        del namespace.inv[addr]
+
+    def iter_addrs(self, name, ns_set=None):
+        """Iterate over address that have a label with a given name in the given set of namespaces. 
+        
+        :param name:    The name of the labels.
+        :param ns_set:  A set of namespaces to work with.
+        :return:        Tuples of (namespace, addr).
+        """
+        ns_set = self._normalize_ns_set(ns_set)
+
+        for ns in ns_set:
+            yield ns, self.get_addr(name, ns)
+
+    #
+    #   ...
+    #
+
+    def _generate_default_label(self, addr):
+        """Generate a default label for the given address.
+        
+        :param addr:    The address to generate a label for.
+        :return:        A default label for the given address.
+        """
+        return 'lbl_%x' % addr
```

Comment:
I would be nice to have different naming schemes based on where the address is and what's at that address (code, data, etc) but that might be for down the line...